fabtests/unit/getinfo_test.c

```diff
@@ -780,6 +780,15 @@ static int getinfo_unit_test(char *node, char *service, uint64_t flags,
 			break;
 	}
 out:
+	/* If init was called, then there is a chance that the hints were
+	 * modified so that the application now owns some of the memory.
+	 * At the moment, only invalid_dom does this and the domain name
+	 * is the only application owned memory. Free the application owned
+	 * memory so that fi_freeinfo only frees memory that it owns. */
+	if (init) {
+		free(test_hints->domain_attr->name);
+		test_hints->domain_attr->name = NULL;
+	}
```

Comment:
Why not just call ft_freehints() here?

Reply:
Because test_hints is created from a call to fi_dupinfo, which makes libfabric the owner of all of it. Then a call to init might make the further change to the domain name.
So the application needs to free the domain name and then everything else needs to be freed by libfabric, thus the call to fi_freeinfo(test_hints) at line 792.

I can refine the comment to be more explicit if you'd like.

Reply:
Thanks - comment is fine.